src/stan/math/prim/arr/meta/is_vector_like.hpp

```diff
@@ -0,0 +1,13 @@
+#ifndef STAN__MATH__PRIM__ARR__META__IS_VECTOR_LIKE_HPP
```

Comment:
Thanks for adding this in. It looks like this never had tests in develop. I'm going to create an issue to create and test `is_vector_like` separately outside of this pull request.